eZ/Publish/Core/Search/Common/FieldNameResolver.php

```diff
@@ -101,17 +92,17 @@ protected function getFieldMap()
                         continue;
                     }
 
-                    $this->fieldTypes[$contentType->identifier][$fieldDefinition->identifier] =
+                    $fieldTypes[$contentType->identifier][$fieldDefinition->identifier] =
                         $fieldDefinition->fieldType;
                 }
             }
         }
 
-        return $this->fieldTypes;
+        return $fieldTypes;
```

Comment:
@andrerom yes, I will add `getFieldMap()` to the `ContentType` handler so it can be SPI cached and invalidated. All (Legacy, Solr and Elasticsearch) engines will benefit from this.